### ISSUE TITLE
pipe-friendly version of re_match?

### ISSUE THREAD

**Issue by mdlincoln** [MangoTheCat/rematch2#10]
Great package! After just a day of using it, I've found myself repeating a very similar pattern, binding the results columns to an input data frame as part of a data cleaning pipe:

``` r
suppressPackageStartupMessages({
  library(rematch2)
  library(tibble)
  library(dplyr)
})
  
model_mtcars <- mtcars %>%
  rownames_to_column() %>% 
  bind_cols(re_match(.$rowname, pattern = "^(?<make>\\w+) (?<model>.+)$"))

str(model_mtcars)
#> 'data.frame':    32 obs. of  16 variables:
#>  $ rowname: chr  "Mazda RX4" "Mazda RX4 Wag" "Datsun 710" "Hornet 4 Drive" ...
#>  $ mpg    : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...
#>  $ cyl    : num  6 6 4 6 8 6 8 4 4 6 ...
#>  $ disp   : num  160 160 108 258 360 ...
#>  $ hp     : num  110 110 93 110 175 105 245 62 95 123 ...
#>  $ drat   : num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...
#>  $ wt     : num  2.62 2.88 2.32 3.21 3.44 ...
#>  $ qsec   : num  16.5 17 18.6 19.4 17 ...
#>  $ vs     : num  0 0 1 1 0 1 0 1 1 1 ...
#>  $ am     : num  1 1 1 0 0 0 0 0 0 0 ...
#>  $ gear   : num  4 4 4 3 3 3 3 4 4 4 ...
#>  $ carb   : num  4 4 1 1 2 1 4 2 2 4 ...
#>  $ make   : chr  "Mazda" "Mazda" "Datsun" "Hornet" ...
#>  $ model  : chr  "RX4" "RX4 Wag" "710" "4 Drive" ...
#>  $ .text  : chr  "Mazda RX4" "Mazda RX4 Wag" "Datsun 710" "Hornet 4 Drive" ...
#>  $ .match : chr  "Mazda RX4" "Mazda RX4 Wag" "Datsun 710" "Hornet 4 Drive" ...
```

I use it so often in data cleaning that I ended up writing a slightly hack-y wrapper function that allows me to chain together several `re_match` calls (removing the `.text` and `.match` columns to avoid dplyr headaches from repeated column names):

```r
pipe_re_match <- function(df, colname, ...) {
  bind_cols(df, select(re_match(text = df[[colname]], ...), -.text, -.match))
}
```

(yes, ideally this would also use non-standard evaluation 😄 )

(and would also ideally not create a dplyr dependency 😛 )

Have you considered adding such a wrapper function?

**Comment by gaborcsardi** [comment]
Yes, I think this would make sense, assuming I can avoid depending on `dlpyr`. We could call it `re_match_column`. 

Do you think it makes sense to support multiple columns with a single function call?

Getting rid of  `.text` is good, but I think `.match` is often useful, so I would keep that, maybe optionally.

**Comment by mdlincoln** [comment]
When you say support multiple columns, do you mean allowing multiple patterns to be evaluated and results added as columns?

As for `.text` and `.match`, I think giving the **option to suppress** either or both would be ideal - if one were calling the function several times consecutively in the pipe, then one would probably want to suppress both. If calling just once, then having both could be very useful.

**Comment by gaborcsardi** [comment]
Well, `.text` is just the input, so if `re_match_column` **adds** the match columns to the data frame, then it is useless to add it. `.match` might be often helpful, though, maybe even by default.

So, do you want to always add the new columns to the input data frame? Because then some other function name would be better, e.g. `bind_re_match`? I am not very familiar with `dplyr` TBH, so bear with me.....

**Comment by mdlincoln** [comment]
Doh, yes, `.text` would be redundant! I wasn't thinking about that.

And yes, the point of the pattern is to always _add_ the columns to the data frame. I think you could just as easily use `cbind` instead of `dplyr::bind_cols`
